### DISCUSSION TITLE
Set annotation font size relative to image dimensions

### DISCUSSION THREAD

**Discussion by trustieee** [dlemstra/Magick.NET#1262] [Help]
I have code like this:

```cs
// add bar at top of image 2.5% of image height
magickImage.Splice(new MagickGeometry(new Percentage(0), new Percentage(2.5)));
// draw text
magickImage.Annotate(annotationText, Gravity.North);
```

But I'd like to set the font size to something relative to the dimensions of the image.
I see in imagemagick docs that not setting a font size will tell imagemagick to automatically scale to try and fit the image, but I don't see that behavior here, or I don't know how to do what I'm trying to with this API.

If there's no way to have it automatically scale, perhaps something based around this?

```cs
magickImage.Settings.FontPointsize = (magickImage.Height * .05) + {some base font size number?};
```

Thanks.

**Comment by dlemstra** [comment]
You could do something like this:

```C#
var settings = new MagickReadSettings
{
    Width = 1000,
    Height = 1000,
    TextGravity = Gravity.Center,
};

using var image = new MagickImage("label:Test", settings);
```

This will put the text "Test" inside the 1000x1000 image and make it as large as possible.

**Comment by trustieee** [comment]
Thanks @dlemstra ,

Edit: unless your intention behind the reply above is that I would overlay that image on top of or inside the spliced area of the original image? Not sure I understand.

I may not be explaining what I'm trying to do correctly. Here's a couple images that may show it better.

Here's a lower resolution image with the spliced white area at the top, and an annotation written inside it (same as using code from my original post)
![bmp1](https://user-images.githubusercontent.com/1885172/192573694-4562f3f0-6745-4f45-b652-2efbc4160833.jpg)

Here's a higher resolution image with the same code, but the font size doesn't scale correctly so it's very tiny
![png1](https://user-images.githubusercontent.com/1885172/192573558-6950eb7f-6405-4254-a15a-bd3cc574a65b.jpg)

Maybe this shows what I'm running into better and can help find a solution.

Thanks!

**Comment by dlemstra** [comment]
Instead of creating a splice you could also calculate those dimensions yourself and create an image of that size with the label example. And then with an `MagickImageCollection` you can append those two images.
